eslint.config.mjs

```diff
@@ -0,0 +1,403 @@
+// @ts-check
+
+import js from '@eslint/js';
+import { globalIgnores } from 'eslint/config';
+import formatjs from 'eslint-plugin-formatjs';
+// @ts-expect-error -- No typings
+import importPlugin from 'eslint-plugin-import';
+import jsdoc from 'eslint-plugin-jsdoc';
+import jsxA11Y from 'eslint-plugin-jsx-a11y';
+import promisePlugin from 'eslint-plugin-promise';
+import react from 'eslint-plugin-react';
+import reactHooks from 'eslint-plugin-react-hooks';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+/** @type {import('typescript-eslint').ConfigArray} */
+export const baseConfig = [
+  js.configs.recommended,
+  importPlugin.flatConfigs.recommended,
+];
+
+export default tseslint.config([
+  baseConfig,
+  globalIgnores([
+    'build/**/*',
+    'coverage/**/*',
+    'db/**/*',
+    'lib/**/*',
+    'log/**/*',
+    'node_modules/**/*',
+    'public/**/*',
+    '!public/embed.js',
+    'spec/**/*',
+    'tmp/**/*',
+    'vendor/**/*',
+    'streaming/**/*',
+  ]),
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  reactHooks.configs['recommended-latest'],
+  jsxA11Y.flatConfigs.recommended,
+  importPlugin.flatConfigs.react,
+  promisePlugin.configs['flat/recommended'],
+  // @ts-expect-error -- For some reason the formatjs package exports an empty object?
+  formatjs.configs.strict,
+  jsdoc.configs['flat/recommended'],
```

Comment:
We probably want the jsdoc plugin in the base config as well?

Reply:
i tried moving it over but it flagged a large amount of issues, and i didn't want to use this already-large PR to fix issues there. i can address in a followup PR or future refactor.

Reply:
odd, looks like it is fine now? i've moved it to base config.